src/main/java/com/google/sps/data/Places.java

```diff
@@ -121,8 +129,15 @@
     Set<PlacesSearchResult> hiddenGems = new HashSet<>();
     hiddenGems = flatten(all_places)
       .filter(place -> 
-        place.rating >= HIDDEN_GEMS_RATINGS_MIN && place.userRatingsTotal >= HIDDEN_GEMS_NUMBER_OF_RATINGS_MIN
-        && place.userRatingsTotal <= HIDDEN_GEMS_NUMBER_OF_RATINGS_MAX)
+        place.rating >= HIDDEN_GEMS_RATINGS_MIN 
+        && place.userRatingsTotal >= HIDDEN_GEMS_NUMBER_OF_RATINGS_MIN
+        && place.userRatingsTotal <= HIDDEN_GEMS_NUMBER_OF_RATINGS_MAX
+        && place.businessStatus.equals(BUSINESS_STATUS_OPERATIONAL)
+        && (
+          (place.types[0].equals(TYPE_CAFE) || place.types[0].equals(TYPE_RESTAURANT))
+          || (place.types[1].equals(TYPE_CAFE) || place.types[1].equals(TYPE_RESTAURANT))
```

Comment:
is this part just checking if the first and second businesses in the string are 'restaurant' or a 'cafe'? why just [0] and [1] elements?

Reply:
Explanation here: https://github.com/googleinterns/step273-2020/pull/79#discussion_r575945715